pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -502,14 +502,13 @@ func getRouteOperation(in *route.Route, vsName string, port int) string {
 		}
 	}
 
-	operation := fmt.Sprintf("%s:%d%s", vsName, port, path)
 	// use host:port + match uri if there is only one destination cluster.
-	wc := in.GetRoute().GetWeightedClusters()
-	if c := wc.GetClusters(); len(c) == 1 {
-		_, _, h, p := model.ParseSubsetKey(c[0].Name)
-		operation = fmt.Sprintf("%s:%d%s", h, p, path)
+	if c := in.GetRoute().GetCluster(); c != "" {
+		_, _, h, p := model.ParseSubsetKey(c)
+		return fmt.Sprintf("%s:%d%s", h, p, path)
```

Comment:
? Please add comments. This entire file is poorly documented, with unintuitive function names that its hard to understand whats going on.

Reply:
GetRouteOperation derives a human-readable descriptor for an envoy route...

Reply:
That much is evident. What is the new change doing?